06-building-a-program-start-to-finish/writing-tests/output/spec.js

```diff
@@ -0,0 +1,159 @@
+/* My Test Setups */
+/* global describe it expect */
+/* global FileReader */
+/* global File */
+/* global InvertedIndex */
+import myBook from './myBook.json';
+import emptyBook from './emptyBook.json';
+import invalidKeys from './invalidBook.json';
+
+const invertedIndex = new InvertedIndex();
+invertedIndex.createIndex('books.json', myBook);
+const getIndex = invertedIndex.getIndex('books.json');
+const jsonFile = new File([JSON.stringify(myBook)],
+  'books.json', { type: 'application/json' });
+
+/* My Suites */
+describe('Class and Method Instantaion', () => {
+  it('Should be instantiated with the new keyword', () => {
+    const init = () => {
+      InvertedIndex();
+    };
+    expect(init).toThrowError('Cannot call a class as a function');
+  });
+
+  it('Should contain the getIndex method', () => {
+    expect(typeof invertedIndex.getIndex).toBe('function');
+  });
+
+  it('Should contain the createIndex method', () => {
+    expect(typeof invertedIndex.createIndex).toBe('function');
+  });
+
+  it('Should contain the searchIndex method', () => {
+    expect(typeof invertedIndex.searchIndex).toBe('function');
+  });
+
+  it('Should contain the readFile method', () => {
+    expect(typeof InvertedIndex.readFile).toBe('function');
+  });
+
+  it('Should contain the tokenize method', () => {
+    expect(typeof InvertedIndex.tokenize).toBe('function');
+  });
+
+  it('Should contain the validateFile method', () => {
+    expect(typeof InvertedIndex.validateFile).toBe('function');
+  });
+});
+describe('InvertedIndex createIndex', () => {
+  it('Should return true for creating Index', () => {
+    expect(invertedIndex.createIndex('books.json', myBook)).toBeTruthy();
+  });
+
+  it('Should return the numbers of books in a file', () => {
+    const numOfBooks = invertedIndex.booksIndex('books.json');
+    expect(numOfBooks.length).toEqual(6);
+  });
+});
+
+describe('InvertedIndex searchIndex', () => {
+  it('Should return `No file has been indexed yet', () => {
+    const search = () => {
+      invertedIndex.searchIndex('alice');
+    };
+    expect(search).toThrowError('Error: No file has been indexed yet');
+  });
+
+  it('Should return an empty if not found', () => {
```

Comment:
Your test specifications are not descriptive enough to communicate what the test is all about.